packages/flame/test/game/transform2d_test.dart

```diff
@@ -170,19 +171,50 @@ void main() {
           ..rotateZ(rotation)
           ..scale(scale.x, scale.y, 1)
           ..translate(offset.x, offset.y);
+
         for (var k = 0; k < 16; k++) {
           expect(
             transform2d.transformMatrix.storage[k],
-            closeTo(matrix4.storage[k], 1e-10),
+            closeTo(
+              matrix4.storage[k],
+              toleranceVector2Float32(translation) +
+                  toleranceFloat32(rotation) +
+                  toleranceVector2Float32(scale) +
+                  toleranceVector2Float32(offset),
+            ),
           );
         }
         // Check round-trip conversion between local and global
         final point1 =
             Vector2((rnd.nextDouble() - 0.5) * 5, (rnd.nextDouble() - 0.5) * 5);
         final point2 =
             transform2d.globalToLocal(transform2d.localToGlobal(point1));
-        expect(point1.x, closeTo(point2.x, 1e-10));
-        expect(point1.y, closeTo(point2.y, 1e-10));
+        expect(
+          point1.x,
+          closeTo(
+            point2.x,
+            (toleranceFloat32(point1.x) +
+                    toleranceFloat32(point2.x) * 2 +
+                    toleranceVector2Float32(translation) +
+                    toleranceFloat32(rotation) +
+                    toleranceVector2Float32(scale) +
+                    toleranceVector2Float32(offset)) *
+                4,
+          ),
+        );
+        expect(
+          point1.y,
+          closeTo(
+            point2.y,
+            (toleranceFloat32(point1.y) +
+                    toleranceFloat32(point2.y) * 2 +
+                    toleranceVector2Float32(translation) +
+                    toleranceFloat32(rotation) +
+                    toleranceVector2Float32(scale) +
+                    toleranceVector2Float32(offset)) *
+                4,
+          ),
+        );
```

Comment:
Would it be easier to use `closeToVector` here?

Reply:
yeah, of course :) I think I broke it out when didn't have the toleranceVector2... helper.